### ISSUE TITLE
Media Upload Status Checks Miss Error Information

### ISSUE THREAD

**Issue by vdms-mark** [linvi/tweetinvi#597]
After uploading a video to Twitter, we call Twitter's Finalize
```
var uploader = Upload.CreateChunkedUploader();
// upload video
var media = uploader.Complete();
```

Afterwards, we can ensure that Twitter's media processing was successful.
`media = Upload.WaitForMediaProcessingToGetAllMetadata(media);`

The problem is that if processing was unsuccessful...
`media.UploadedMediaInfo.ProcessingInfo.State == "failed"`
the error message related to the failure is not accessible.  According to Twitter's documentation, an error object is included in the response, indicating what when wrong.  The easiest way to reproduce this type of failure is to upload a video longer than 140 seconds.

https://developer.twitter.com/en/docs/media/upload-media/api-reference/get-media-upload-status
Sample error response from the documentation:
```
// Example of a failed response:

{
  "media_id":710511363345354753,
  "media_id_string":"710511363345354753",
  "processing_info":{
    "state":"failed",
    "progress_percent":12,
    "error":{
      "code":1,
      "name":"InvalidMedia",
      "message":"Unsupported video format"
    }
  }
}
```

As a workaround, I am using the TwitterAccessor class to do a custom query for the media status, and I just parse the returned JSON. However, I think it makes sense if this type of error information is provided natively by the library.

**Comment by linvi** [comment]
I am not sure what I will do with this one. Would you expect the exception to be raised containing the error message?

**Comment by linvi** [comment]
Mmm or maybe directly within the processing info.

**Comment by linvi** [comment]
This is now done. Please check a new media upload process example that will be available in 2.2.

``` c#
var binary = File.ReadAllBytes("./video.mp4");

Console.WriteLine("Starting upload...");

var media = Upload.UploadVideo(binary, UploadMediaCategory.TweetVideo, progressChanged =>
{
    switch (progressChanged.State)
    {
        case UploadProgressState.INITIALIZED:
            Console.WriteLine("Media Upload initialized with Twitter");
            break;
        case UploadProgressState.PROGRESS_CHANGED:
            Console.WriteLine($"Progress : {progressChanged.Percentage}%");
            break;
        case UploadProgressState.COMPLETED:
            Console.WriteLine("Media Uploaded, please wait for Twitter to complete its processing");
            break;
    }
});

Upload.WaitForMediaProcessingToGetAllMetadata(media);

if (media.IsReadyToBeUsed)
{
    Console.WriteLine("The media ready to be used!");

    var tweet = Tweet.PublishTweet("hello", new PublishTweetOptionalParameters()
    {
        Medias = { media }
    });
}
else
{
    var errorInfo = media.UploadedMediaInfo.ProcessingInfo.Error;
}
```
